examples/all-clusters-app/linux/AllClustersCommandDelegate.cpp

```diff
@@ -228,14 +229,20 @@ void AllClustersCommandDelegate::OnSwitchEventHandler(uint32_t eventId)
 
     if (eventId == Clusters::Switch::Events::SwitchLatched::Id)
     {
+        EmberAfStatus status = Switch::Attributes::CurrentPosition::Set(endpoint, newPosition);
+        VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(NotSpecified, "Failed to set CurrentPosition attribute"));
         Clusters::SwitchServer::Instance().OnSwitchLatch(endpoint, newPosition);
     }
     else if (eventId == Clusters::Switch::Events::InitialPress::Id)
     {
+        EmberAfStatus status = Switch::Attributes::CurrentPosition::Set(endpoint, newPosition);
+        VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(NotSpecified, "Failed to set CurrentPosition attribute"));
         Clusters::SwitchServer::Instance().OnInitialPress(endpoint, newPosition);
     }
     else if (eventId == Clusters::Switch::Events::LongPress::Id)
     {
+        EmberAfStatus status = Switch::Attributes::CurrentPosition::Set(endpoint, newPosition);
+        VerifyOrReturn(EMBER_ZCL_STATUS_SUCCESS == status, ChipLogError(NotSpecified, "Failed to set CurrentPosition attribute"));
         Clusters::SwitchServer::Instance().OnLongPress(endpoint, newPosition);
     }
     else if (eventId == Clusters::Switch::Events::ShortRelease::Id)
```

Comment:
Release should reset the position, no?

Reply:
Yes, it is better to reset the currentPosition to 0 during release